### PR TITLE
h2oGPT Helm Chart

### PR DESCRIPTION
**Pull request by EshamAaqib**
## Description 

Helm Chart for h2oGPT, which includes vLLM and TGI

### CODE REVIEW

**Review by pseudotensor — APPROVED** [review]
great!

**Review by ozahavi — APPROVED** [review]
lgtm

File: helm/h2ogpt-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+              port: http
+            {{- toYaml .Values.vllm.readinessProbe | nindent 12 }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.vllm.resources | nindent 12 }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "h2ogpt.fullname" . }}-vllm-inference-config
+          env:
+          - name: TRANSFORMERS_CACHE
```
Comment (achraf-mer) [inline helm/h2ogpt-chart/templates/deployment.yaml:338]: `TRANSFORMERS_CACHE` not really needed, we noticed that sometimes when setting env vars, it ends up having side effects different than what we expect, so best to keep the defaults if not necessary to change.

See an example here: https://github.com/h2oai/h2ogpt/issues/755#issuecomment-1696734046

**Comment by babytdream** [comment]
@EshamAaqib @pseudotensor @ozahavi @achraf-mer @arnocandel 
Hi, are you interested in this question about improving vllm and benefiting you in #794 ？

**Pull request merged**